hasher-matcher-actioner/hmalib/common/message_models.py

```diff
@@ -89,25 +95,76 @@ def from_match_message_action_label_and_action_rules(
 
 
 @dataclass
-class WritebackMessage(MatchMessage):
+class WritebackMessageConfig:
     """
+    Simple holder for getting typed environment variables
+    """
+
+    writebacks_queue_url: str
+    sqs_client: SQSClient
+
+    @classmethod
+    @lru_cache(maxsize=None)
+    def get(cls):
+        HMAConfig.initialize(os.environ["CONFIG_TABLE_NAME"])
+        return cls(
+            writebacks_queue_url=os.environ["WRITEBACKS_QUEUE_URL"],
+            sqs_client=boto3.client("sqs"),
+        )
+
+
+@dataclass
+class WritebackMessage(HasAWSSerialization):
+    """
+    Writebacks happen on a collection of BankedSignals. To perform a write back,
+    instantiate an instacne of this class and run the send_to_queue method
+
     The Writebacker needs the match message plus which writeback type to perfrom
     on the source of the signal (for now, ThreatExchange).
     """
 
+    banked_signals: t.List[BankedSignal]
+
     writeback_type: WritebackTypes.WritebackType = field(
-        default=WritebackTypes.UnspecifiedWriteback
+        default=WritebackTypes.NoWriteback
     )
 
     @classmethod
-    def from_match_message_and_label(
+    def from_match_message_and_type(
         cls,
         match_message: MatchMessage,
         writeback_type: WritebackTypes.WritebackType,
     ) -> "WritebackMessage":
         return cls(
-            match_message.content_key,
-            match_message.content_hash,
             match_message.matching_banked_signals,
             writeback_type,
         )
+
+    @classmethod
+    def from_banked_signal_and_opinion_change(
+        cls, banked_signal: BankedSignal, opinion_change: PendingOpinionChange
+    ) -> "WritebackMessage":
+        opinion_change_to_writeback_type = {
+            PendingOpinionChange.MARK_TRUE_POSITIVE: WritebackTypes.TruePositive,
+            PendingOpinionChange.MARK_FALSE_POSITIVE: WritebackTypes.FalsePositive,
+            PendingOpinionChange.REMOVE_OPINION: WritebackTypes.RemoveTrueFalsePositive,
```

Comment:
Why not `RemoveOpinion` to match the enum? I'm not tied to that name but `RemoveTrueFalsePositive` is a little too confusing imo.

Reply:
I was thinking that "Opinion" is too general and "Ingested" or "Seen" wiritebacks might be considered and opinion. However I think you're right that Opinion is clearer